docs/lang_spec/Cadence Language Specification.md

```diff
@@ -0,0 +1,5798 @@
+
+# Cadence Language Specification
+
+**THIS IS THE START OF A SPECIFICATION**
+**: It is incomplete and may contain many errors.**
+
```

Comment:
It might be good to break this document down into multiple files, as it is already very large, which makes it difficult to review (GitHub is already slowing down 😅)

Reply:
Agree. Given that markdown doesn't support a way of embedding content from a different file, maybe we can still have them on separate files, but combine them later using a third-party tool, and generate the "single-file version" automatically (for example, during the build).

Reply:
Yeah, not having a single file at the beginning is OK, making it easier to write is more important.
For now we can just use the Markdown links to link to other files.

I had quick look around regarding tools, some ideas:
- Do nothing / don't generate anything from the Markdown files and keep them as is. Content can be viewed on GitHub, which has page previews
- Use [Docusaurus](https://docusaurus.io), just like we use for the website/documentation on https://cadence-lang.org / https://github.com/onflow/cadence-lang.org. We might also just move or build the spec there. There seem to be plugins to generate PDF
- The [Kotlin Spec](https://github.com/Kotlin/kotlin-spec/) uses a pre-processor, https://logological.org/gpp, to stitch the files together, and [pandoc](https://pandoc.org/) to generate HTML and PDF. See e.g. https://github.com/Kotlin/kotlin-spec/blob/release/docs/scripts/build/buildPdf.sh and https://github.com/Kotlin/kotlin-spec/blob/release/docs/scripts/build/buildHtml.sh